internal/pkg/dir/dir_test.go

```diff
@@ -57,3 +59,56 @@ func TestInit(t *testing.T) {
 	}
 	checkPermissions(t, bar, 0755)
 }
+
+func TestCopy(t *testing.T) {
+	src := t.TempDir()
+	srcDirname := path.Base(src)
+
+	// Test that empty directories don't copy anything
+	dst := t.TempDir()
+	err := Copy(src, dst)
+	if err != nil {
+		t.Fatalf("Unable to copy empty dir: %v", err)
+	}
+
+	//rmdir will fail if the directory has anything at all
+	err = exec.Command("rmdir", path.Join(dst, srcDirname)).Run()
+	if err != nil {
+		t.Fatalf("Unable to remove supposedly empty dir: %v", err)
+	}
+
+	// Test that copying actually works
+	dst = t.TempDir()
+	expectedDirs := []string{"dir1/", "dir2/", "dir2/dir1/", "dir2/dir2/"}
+	expectedFiles := []string{"dir1/file1", "dir1/file2", "dir2/file1", "dir2/dir2/file1"}
+
+	for _, dir := range expectedDirs {
+		p := path.Join(src, dir)
+		err = exec.Command("mkdir", p).Run()
+		if err != nil {
+			t.Fatalf("Unable to create directory %s: %v", p, err)
+		}
+	}
+
+	for _, file := range expectedFiles {
+		p := path.Join(src, file)
+		err = exec.Command("touch", p).Run()
```

Comment:
Do you want to create those files? I'd just do sth. like `os.WriteFile(p, []byte{}, 0644)` instead.